rasp/base.py

```diff
@@ -61,9 +75,10 @@ def get_page_source(self, url, params=None, headers=None):
         """
         if not url:
             raise ValueError('url needs to be specified')
-        if isinstance(headers, dict):
-            temp = headers
-            headers = deepcopy(self.headers)
+
+        temp = headers
+        headers = deepcopy(self.headers)
+        if isinstance(temp, dict):
             headers.update(temp)
```

Comment:
I'm an idiot.  I wrote this code and I didn't realize I created a circular reference.

Look at it a little closer, the operation ends up being `deepcopy(self.headers).update(deepcopy(self.headers))`.  Instead, let's try something like this:
```python
merged_headers = deepcopy(self.headers)
if isinstance(headers, dict):
    merged_headers.update(headers)

response = self.session.get(
    url, params=params, headers=merged_headers
)
```